pkg/apis/v1/ec2nodeclass_conversion.go

```diff
@@ -104,33 +123,81 @@ func (in *EC2NodeClass) ConvertFrom(ctx context.Context, from apis.Convertible)
 	v1beta1enc := from.(*v1beta1.EC2NodeClass)
 	in.ObjectMeta = v1beta1enc.ObjectMeta
 
-	// TODO: jmdeal@ remove before v1
-	// Temporarily fail closed when trying to convert EC2NodeClasses with the Ubuntu AMI family since compatibility support isn't yet integrated.
-	// This check can be removed once it's added.
-	if lo.FromPtr(v1beta1enc.Spec.AMIFamily) == v1beta1.AMIFamilyUbuntu {
-		return fmt.Errorf("failed to convert v1beta1 EC2NodeClass to v1, conversion for Ubuntu AMIFamily is currently unsupported")
-	}
+	switch lo.FromPtr(v1beta1enc.Spec.AMIFamily) {
```

Comment:
The reason they're not is the dependency on the annotation. We could move it if we either passed down the annotation map, or returned up a map of annotations, but it seems cleaner to me to keep logic that spans these substructures in the top-level function.

Reply:
The main reason I'm leaning not to is because at that point, for consistencies sake, we'd want to move the block handling this conversion in convert from into the spec conversion helper. I'm not personally a fan of how that would look as that would involve either:
a. Passing in and mutating the annotation map directly
b. Returning the set of annotations from the function signature. 

Since these mutations work across the top-level fields, I think it makes since to keep it in the top level function. IMO dropping the helper functions altogether makes sense but I didn't have a reason to do that.